tweetapi/authentication.py

```diff
@@ -0,0 +1,34 @@
+from rest_framework.authentication import get_authorization_header,BaseAuthentication
+from rest_framework import exceptions
+from django.contrib.auth.models import User
+import jwt
+from django.conf import settings
+
+class Authentication(BaseAuthentication):
+    def authenticate(self, request):
+        auth_header = get_authorization_header(request)
+
+        auth_data = auth_header.decode('utf-8')
+        auth_token = auth_data.split(" ")
+
+        if len(auth_token) !=2:
+            raise exceptions.AuthenticationFailed("TOKEN NOT VALID")
+        
+        token = auth_token[1]
+
+        try:
+            payload = jwt.decode(token,"secret",algorithms=["HS256"])
+            id = payload['id']
+            user = User.objects.get(id=id)
```

Comment:
these two can be in one line. Also consider a case where id is not present in payload